lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -123,6 +143,27 @@ class UserProfileBloc {
     return user;
   }
 
+  void _listenUserActions() {
+    _userActionsController.stream.listen((action) {
+      var handler = _actionHandlers[action.runtimeType];
+      if (handler != null) {
+        handler(action).catchError((e) => action.resolveError(e));
+      }
+    });
+  }
+
+  Future _updateSecurityModel(UpdateSecurityModel updateSecurityModelAction) async {
+    if (updateSecurityModelAction.pinCode != null && _currentUser.securityModel.pinCode != updateSecurityModelAction.pinCode) {
```

Comment:
This is buggy since if the new pinCode equals the current pinCode we will delete the pinCode.